lib/GADS/Datum/File.pm

```diff
@@ -153,10 +153,30 @@ sub _build_files
     elsif ($self->has_ids) {
         @return = $self->_ids_to_files(@{$self->ids});
     }
+    
+    return [+{
+        id => -1,
+        name => 'Purged',
+        mimetype => 'text/plain',
+    }] if $self->is_purged;
 
     return \@return;
 }
 
+sub _datums
+{   my $self = shift;
+    [$self->schema->resultset('File')->search({
+        record_id => $self->record_id,
+        layout_id => $self->column->id,
+    })->all];
+}
+
+sub is_purged {
+    my $self = shift;
+    my @datums = @{$self->_datums};
+    return grep { $_->purged_on } @datums;
```

Comment:
This should use `is_purged` in order to keep things consistent with other classes

Reply:
This is because files are handled differently, or am I having a "moment"?